### PR TITLE
Created and implemented a Refinery and CoolingTower recipe API

### PR DESCRIPTION
**Pull request by iTitus**

### CODE REVIEW

File: src/main/java/buildcraftAdditions/recipe/duster/DusterRecipe.java [hunk 1]
```diff
@@ -27,14 +27,16 @@ public DusterRecipe(ItemStack input, ItemStack output) {
...
 	}
 
 	@Override
 	public ItemStack getOutput(ItemStack input) {
-		return Utils.areItemStacksEqualItem(input, this.input) ? output.copy() : null;
+		return Utils.areItemStacksEqualItem(this.input, input) ? output.copy() : null;
 	}
 
+	@Override
+	public String toString() {
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/recipe/duster/DusterRecipe.java]: this is unneeded

Comment (iTitus) [inline src/main/java/buildcraftAdditions/recipe/duster/DusterRecipe.java]: Oooops, fail

File: src/main/java/buildcraftAdditions/tileEntities/TileRefinery.java [hunk 2]
```diff
@@ -100,14 +98,13 @@ private void updateHeat() {
...
 			requiredHeat = recipe.getRequiredHeat();
 			lastRequiredHeat = requiredHeat;
-			outputFluid = recipe.getOutput();
-			outputAmount = recipe.getOutputAmount();
-			inputAmount = recipe.getInputAmount();
+			outputFluidStack = recipe.getOutput();
+			inputFluidStack = recipe.getInput();
 		} else
-		requiredHeat = 0;
+			requiredHeat = 0;
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/tileEntities/TileRefinery.java]: why the extra indentation here?

Comment (iTitus) [inline src/main/java/buildcraftAdditions/tileEntities/TileRefinery.java]: I dont know :P

**Pull request merged**